### PR TITLE
[1LP][WIP]Fix for test_permissions_catalog_add.

### PR DESCRIPTION
**Pull request by anurag03**
Purpose or Intent
=================
There were issues with the role creation and unnecessary argument passing to catalog.create(), which is fixed in this PR.

{{pytest: -v -k 'test_permissions_catalog_add or test_permissions'}}

### CODE REVIEW

File: cfme/tests/services/test_catalog.py [hunk 1]
```diff
@@ -31,6 +34,9 @@ def test_catalog_duplicate_name():
     cat.delete()
 
 
+# Un-collecting for 5.8 because of BZ: 1486041
+@pytest.mark.uncollectif(lambda: current_version() < '5.9')
```
Comment (izapolsk) [inline cfme/tests/services/test_catalog.py]: REQUIRED: please use BZ instead

NB: please don't mix several fixes in one PR

Comment (anurag03) [reply cfme/tests/services/test_catalog.py]: @izapolsk  actually this is a fix for 5.9, but in 5.8 this test case fails due to above BZ.
The problem with the above BZ is that it's marked as verified, so here BZ can't be used.

Comment (izapolsk) [reply cfme/tests/services/test_catalog.py]: @anurag03, 
1. there should be clone of this BZ for 5.8 in Bugzilla then. please ask Simaishi about clone.
2. replace current_version with  lambda: store.current_appliance.version

**Comment by anurag03** [comment]
@izapolsk : Closing this PR as the fix is already provided by some other PR.

**Pull request closed without merging**